### ISSUE TITLE
[BUG] Ghost enemies are solid on Odamex

### ISSUE THREAD

**Issue by NinjaDelphox** [odamex/odamex#806]
**Describe the bug**
I have a few dehacked/decorate "ghost" enemies in a mapset, and they are solid only on Odamex when they should be able to just be walked through.

**Build that the bug occurred in**
Odamex V10.2.0 (gf6615-8433)

**To Reproduce**
To reproduce these monsters, you give them the following attributes in dehacked/decorate:

Height = 65536 (1)
NOGRAVITY+SHOOTABLE+FLOAT+TRANSLUCENT+SOLID+NOCLIP+INFLOAT+COUNTKILL
(One of these monsters is an Arch Vile, so NOGRAVITY+FLOAT+INFLOAT would be removed for it)
All the other attributes are exactly the same as a vanilla Cacodemon/Arch-Vile.

**Expected behavior**
They should behave as a true vanilla ghost monster, as if they were resurrected from a crushed corpse, being completely non-solid and being able to walk through walls. These monsters however are entirely solid on Odamex. 

A somewhat related issue, enemies that are not flagged as solid get stuck if spawned inside another solid object, which does not occur on any other port I have used to test. (DSDA-Doom 0.24.3, Eternity 4.0.3 Devbuild, Zandronum 3.1)

**Additional context**
These problems do not appear on the mentioned ports (and presumably the most recent GZDoom). These monsters appear in just two maps in this megawad, one of which being in the map Hellcinerator. (Which has an issue with running at all on this port.)

**Comment by rakohus** [comment]
Thanks @NinjaDelphox. Could you provide a sample wad?

**Comment by NinjaDelphox** [comment]
Yes, here is a link to the map using the monsters and assets required to play it:

https://drive.google.com/file/d/1O0OHwjN84MakoBqGihblnE1uSp3z6buX/view?usp=sharing

The thing ID's for the ghost monsters are:
Ghost Cacodemon: #8006
Ghost Venomous Cacodemon: #8020
Ghost Arch Vile: #8012

**Comment by elf-alchemist** [comment]
As the drive link above is busted, NinjaDelphox was asked on the discord server itself, the level to test is map 27 of Corruption:
https://www.doomworld.com/forum/topic/133225-corruption-35-map-mbf21-megawad-back-on-idgames

**Comment by hiSandog** [comment]
The compatibility rule should be isolated to the vanilla crushed-corpse ghost state rather than making every actor with `SOLID+NOCLIP` non-blocking. A regression map can spawn the provided thing ids plus a normal noclip actor and compare movement/collision before and after resurrection. Checking the same fixture against the target MBF21/DEH behavior would pin down which flag/state transition Odamex currently interprets differently.

**Comment by bcahue** [comment]
The ghost cacos in MAP15 of corruption.wad seem to be exhibiting correct behavior, but I haven't tested with `co_novileghost` on/off as that may be making them unsolid.
